frontend/datagathering.cpp

```diff
@@ -131,6 +131,7 @@ void MainWindow::on_instrumentsBox_currentTextChanged(const QString &arg1)
 {
     cachedMassSearch_.clear();
 
+    ui_->instrumentButton->setText(arg1.toUpper());
```

Comment:
This actually raises a small issue that we can deal with in the instrument XML file. While most display names for instruments are just the uppercase equivalents as you use here, there is at least one which doesn't conform to that (`SANS2d`).  We can store the "display name" of the instrument in our XML file to give the people what they want.  For sure the NIMROD guy is really picky about his instrument name being correctly capitalised...